src/ui/server/controllers/insurance/business/company-details/index.ts

```diff
@@ -142,9 +150,57 @@ const postCompaniesHouseSearch = async (req: Request, res: Response) => {
   }
 };
 
+/**
+ * saves and goes back to all sections from company details page unless there are database errors
+ * @param req
+ * @param res
+ * @param {Express.Request} Express request
+ * @param {Express.Response} Express response
+ * @returns {Express.Response.redirect} redirects to all sections page on success
+ */
+const postCompanyDetailsSaveAndBack = async (req: Request, res: Response) => {
+  try {
+    const { application } = res.locals;
+
+    const { referenceNumber } = req.params;
+
+    if (!application) {
+      return res.redirect(ROUTES.PROBLEM_WITH_SERVICE);
+    }
+
+    const { body } = req;
+    // runs companiesHouse validation and api call first for companiesHouse input
+    const response = await companiesHouseSearch(body);
+    let { validationErrors } = response;
+    const { company } = response;
+
+    // run validation on other fields on page
+    validationErrors = companyDetailsValidation(body, validationErrors);
+
+    // body for update containing companies house info and request body
+    const updateBody = {
+      ...body,
+      ...company,
+    };
+
+    // runs save and go back commmand
+    const saveResponse = await mapAndSave.companyDetailsSave(updateBody, application, validationErrors);
+
+    if (!saveResponse) {
+      return res.redirect(ROUTES.PROBLEM_WITH_SERVICE);
+    }
+
+    // redirect to all sections page
+    return res.redirect(`${INSURANCE_ROOT}/${referenceNumber}${ALL_SECTIONS}`);
+  } catch (error) {
+    console.error('Error posting save and back company details', { error });
+    return res.redirect(ROUTES.PROBLEM_WITH_SERVICE);
+  }
+};
+
```

Comment:
Can we move this into it's own controller? Pretty big now. in other places have this structure

```js
about-goods-or-services/index.ts < GET and POST
about-goods-or-services/save-and-back/index.ts < POST
```

Reply:
done